blockchain/chain.go

```diff
@@ -781,24 +828,32 @@ func (b *BlockChain) calcSequenceLock(tx *btcutil.Tx, utxoView *UtxoViewpoint,
 		case sequenceNum&RelativeLockSeconds == RelativeLockSeconds:
 			// This input requires a relative time lock expressed
 			// in seconds before it can be spent. Therefore, we
-			// need to query for the block in which this input was
-			// included within so we can compute the past median
-			// time for the block prior to the one which included
-			// this referenced output.
+			// need to query for the block prior to the one in
+			// which this input was included within so we can
+			// compute the past median time for the block prior to
+			// the one which included this referenced output.
 			// TODO: caching should be added to keep this speedy
-			var blockNode *blockNode
-			if err := b.db.View(func(dbTx database.Tx) error {
-				header, err := dbFetchHeaderByHeight(dbTx, inputHeight-1)
-				if err != nil {
-					return err
-				}
-				blockSha := header.BlockHash()
-				blockNode = newBlockNode(header, &blockSha, inputHeight)
-				return nil
-			}); err != nil {
-				return nil, err
+			inputDepth := uint32(b.bestNode.height-inputHeight) + 1
+			blockNode, err := b.ancestorNode(b.bestNode, inputDepth)
+			if err != nil {
+				return sequenceLock, err
+			}
+
+			// The call above loaded all the headers between the
+			// inclusionary block and the chain tip. In order to
+			// properly calculate the proper relative lock time, we
+			// also need to load into memory enough blocks from the
+			// MTP calculation.
+			_, err = b.ancestorNode(b.bestNode, medianTimeBlocks)
```

Comment:
Yep you're right, upon a closer look this additional call to the `ancestorNode` function isn't needed. Also calling the function with `b.bestNode` rather than `blockNode` is a typo/bug. 

I've removed the unnecessary (and incorrect) call to `ancestorNode`.